src/components/AddUser.tsx

```diff
@@ -1,10 +1,42 @@
 import { Box, Button, Checkbox, FormControlLabel, TextField } from '@mui/material'
-import React, { FC } from 'react'
+import React, { FC, useState } from 'react'
+import { useNavigate } from 'react-router'
+import { useCreateUser } from '../services/addUser'
+import { User } from '../utils/types'
 
 const AddUser: FC = () => {
 
+    const [addUser] = useCreateUser()
+    const [user, setUser] = useState<Omit<User, 'id'>>({
```

Comment:
It's better to separate the usage of useState for each property instead of dealing with the whole objects. Please, take a look at this article https://blog.thoughtspile.tech/2021/10/11/usestate-object-vs-multiple/